src/main/java/gov/nasa/jpf/constraints/solvers/smtinterpol/TermParser.java

```
@@ -190,7 +191,44 @@ private void getFirstLet() {
         int idx = this.input.indexOf("(let");
         String prefix = this.input.substring(0, idx);
         this.input = this.input.substring(idx);
+        String postfix = preparePostfix();
         parseOneLet();
-        this.input = prefix + this.input;
+        this.input = prefix + this.input + postfix;
+    }
+
+    private String preparePostfix(){
+        int endIdx = getEndOfLetTerm(this.input, 5,0)+1;
+        String postfix = this.input.substring(endIdx);
+        postfix = postfix.trim();
+        this.input = this.input.substring(0,endIdx);
+        //In case this was not the last let in the term, we need to remove the closing bracket.
+        endIdx = getEndOfLetTerm(postfix, 0,-1);
+        postfix = postfix.substring(0,endIdx) + postfix.substring(endIdx+1);
+        return postfix;
+    }
+    private int getEndOfLetTerm(String analysisPart, int start, int breakCondition){
+        int open = 0;
+        int closed = 0;
+        char[] inputs = analysisPart.toCharArray();
+        //index 5 is the first opening break after (let 
+        for(int index = start ; index < inputs.length; index++){
+          char c = inputs[index];
+          if(c == '('){
+            ++open;
+          }
+          if(c == ')'){
+            ++closed;
+          }
+          if(open-closed == breakCondition){
+            return index;
+          }
+        }
+        System.err.println("Cannot fin the end of the let term");
+        System.exit(42);
```

We should throw an exception here instead of terminating.